src/modules/app/StateType.js

```diff
@@ -2,49 +2,50 @@
 
 import type { StoreActionType } from './StoreActionType'
 import type { PersistState } from 'redux-persist/src/types'
-import { CategoryModel, CityModel, LanguageModel } from '@integreat-app/integreat-api-client'
-import type { ResourceCacheType } from '../endpoint/ResourceCacheType'
-
-export type LanguageStateType = string | null
-
-export type CurrentCityStateType = string | null
+import { CategoryModel, CityModel, EventModel, LanguageModel } from '@integreat-app/integreat-api-client'
+import Moment from 'moment'
 
 type PathType = string
 
-export type RouteStateType = {
-  +root: ?string,
+export type CategoryRouteStateType = {|
+  +root: ?string, // path of the root category
   +depth: number,
-  +models: { [path: PathType]: CategoryModel }, /* Models could be stored outside of RouteStateType
+  +models: { [path: PathType]: CategoryModel }, /* Models could be stored outside of CategoryRouteStateType
                                                    (e.g. CategoriesStateType) to save memory
                                                    in the state. This would be an optimization! */
   +children: { [path: PathType]: Array<PathType> }
-}
+|}
+
+export type EventRouteStateType = {|
+  +path: string | null,
+  +models: Array<EventModel>
```

Comment:
Why would we need multiple events?

Reply:
There is the initial view which shows the overview of all events